### PR TITLE
Improve documentation regarding artifacthub.io annotations

### PR DESCRIPTION
**Pull request by tongpu**
# Description

During my previous PR I've realized that the documentation for the `artifacthub.io` annotations could be improved.

# Checklist

* [x] This PR contains a description of the changes I'm making
* [ ] I updated the version in Chart.yaml
* [ ] I updated the changelog according to the `artifacthub.io/changes` annotation in the Chart.yaml. See following [example](https://github.com/adfinis/helm-charts/pull/858/files).
* [ ] I updated applicable README.md files using  `pre-commit run`
* [ ] I documented any high-level concepts I'm introducing in `docs/`
* [ ] CI is currently green and this is ready for review
* [ ] I am ready to test changes after they are applied and released

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -161,3 +161,22 @@ After installing the pre-commit requirements you can initialize pre-commit.
 pre-commit install
 pre-commit install-hooks
 ```
+
+### Changelog
+
+We're adding [Artifact Hub annotations](https://artifacthub.io/docs/topics/annotations/helm/) to our charts to improve the visibility of changes for our charts. These annotations are added to `Chart.yaml` as a YAML string and need to be replaced on every update, as they are not additive but specific to this release.
```
Comment (hairmare) [inline docs/development.md:167]:
```suggestion
We add [Artifact Hub annotations](https://artifacthub.io/docs/topics/annotations/helm/) to our charts to improve the visibility of changes. These annotations are added to `Chart.yaml` as a YAML string and need to be replaced on every update, as they are not additive but specific to this release.
```

File: docs/development.md [hunk 2]
```diff
@@ -161,3 +161,22 @@ After installing the pre-commit requirements you can initialize pre-commit.
...
+
+```yaml
+annotations:
+  artifacthub.io/changes: |
+    - kind: changed
+      description: "feat(argo-rollouts): Upgrade to argo-rollouts 1.3.0"
```
Comment (hairmare) [inline docs/development.md:173]: is this a good example? it's lazy copypasta from upstreams git history but also highly redundant. maybe our example could be more concise:
```suggestion
      description: "feat(argo-rollouts): Update to 1.3.0"
```
or even just this:
```suggestion
      description: "argo-rollouts: Update to 1.3.0"
```

File: docs/development.md [hunk 3]
```diff
@@ -161,3 +161,22 @@ After installing the pre-commit requirements you can initialize pre-commit.
...
+      links:
+        - name: GitHub PR
+          url: https://github.com/argoproj/argo-helm/pull/1466
+    - kind: changed
+      description: "fix(argo-rollouts): Upgrade argo-rollouts to 1.3.1"
+      links:
+        - name: GitHub PR
```
[the remark this answers is not in the document]
Comment (tongpu) [reply docs/development.md:180]: I think I'm going to extend the examples with cases for both.

File: docs/development.md (hunk 1 above)
Comment (tongpu) [reply docs/development.md]: Yes, definitely better like that.

File: docs/development.md (hunk 2 above)
Comment (tongpu) [reply docs/development.md]: What do you think about the examples now?

**Review by hairmare — APPROVED** [review]

**Pull request merged**